File: Lambda/Model/src/main/scala/prv/zielony/transaction/analyst/lambda/model/Transfer.scala
package prv.zielony.transaction.analyst.lambda.model

/**
 * Created by zielony on 15.02.16.
 */
trait Transfer {

  val id:Long;

  val sourceAccount:Account;

  val targetAccount:Account;

  val amount:Double;

  val currency:Currency;
}

File: Classical/src/main/java/prv/zielony/transaction/analyst/classical/model/Transfer.java
package prv.zielony.transaction.analyst.classical.model;

import javax.persistence.DiscriminatorValue;
import javax.persistence.Entity;
import javax.persistence.OneToOne;

/**
 * Created by zielony on 26.02.16.
 */
@Entity
@DiscriminatorValue("TRANSFER")
public class Transfer extends Transaction {

    @OneToOne
    private Account sourceAccount;

    @OneToOne
    private Account targetAccount;

    public Account getSourceAccount() {
        return sourceAccount;
    }

    public void setSourceAccount(Account sourceAccount) {
        this.sourceAccount = sourceAccount;
    }

    public Account getTargetAccount() {
        return targetAccount;
    }

    public void setTargetAccount(Account targetAccount) {
        this.targetAccount = targetAccount;
    }
}